### ISSUE TITLE
presentFullscreenPlayer() crashes iOS app with controls={true} [re-open #2808]

### ISSUE THREAD

**Issue by linfan68** [TheWidlarzGroup/react-native-video#3025]
# Bug

App crashes with:

Application tried to present modally a view controller <react_native_video.RCTVideoPlayerViewContgroller:......> that has a parent view controller <UIViewController:....>

NOTICE：
**Without controls={true}, it seems working fine**

## Platform

Which player are you experiencing the problem on:
* iOS

## Environment info

React native info output:

```bash
 System:
    OS: macOS 13.1
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Memory: 595.77 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.19.0 - /var/folders/vl/vdw5g5015dx0g0zbdzqy0ckr0000gp/T/yarn--1675594097687-0.6748312658238156/node
    Yarn: 1.22.19 - /var/folders/vl/vdw5g5015dx0g0zbdzqy0ckr0000gp/T/yarn--1675594097687-0.6748312658238156/yarn
    npm: 8.19.3 - /usr/local/bin/npm
    Watchman: 2023.01.09.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0 
    react-native: 0.70.5 => 0.70.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version:  latest master (#dc8c618d41b664afd2c90f172cf269c1a805bb37)

## Steps To Reproduce

1. call videoRef.current?.presentFullscreenPlayer()
...

## Expected behaviour

1. Player enter full screen mode

## Reproducible sample code

```tsx
const TestAutpPlayPostList = () => {
  const videoRef = useRef<Video | null>(null)
  const [isFullscreen, setIsFullscreen] = useState(false)
  useEffect(() => {
    if (isFullscreen) {
      videoRef.current?.presentFullscreenPlayer()
    }
  }, [isFullscreen])
  return <TouchableHighlight onPress={() => {
    console.log('press')
    setIsFullscreen(true)
  }} style={[dbgBorder('blue'), { height: '100%' }]}>
    <Video
      style={{ width: 500, height: 500 }}
      controls={true}
      bufferConfig={{
        maxBufferMs: 10000,
        minBufferMs: 5000,
      }}
      resizeMode='contain'
      fullscreen={false}
      ref={videoRef}
      source={{ uri: 'https://bigger-qyyd.oss-cn-beijing.aliyuncs.com/users/63cb584f1f4e9bfa348bbed2/posts/SVID_20230130_204929_1.mp4' }}
      onLoad={() => console.log('onLoad')}
      onEnd={() => console.log('onEnd')}
      paused={false}
      repeat={false}
      muted={true}
    />
  </TouchableHighlight>
}
```

## Video sample
https://bigger-qyyd.oss-cn-beijing.aliyuncs.com/users/63cb584f1f4e9bfa348bbed2/posts/SVID_20230130_204929_1.mp4

**Comment by clementnv** [comment]
Any update ?

**Comment by freeboub** [comment]
I just open PR which shall fix the issue here : https://github.com/react-native-video/react-native-video/pull/3230
Can you please test it on your side ?

**Comment by nicolaosm** [comment]
Thanks @freeboub! I'll test tomorrow on few ios devices and simulators and report back if its fixed.

**Comment by GhayoorUlHaq** [comment]
I tested that PR but the issue is when we hide the default controls with `controls={false}` and move to full screen, we can still see the default controls.

I can see this [issue](https://github.com/react-native-video/react-native-video/issues/2886) the comments inside **setFullscreen** from `RCTVideo.swift`
- `// Controls will be displayed even if it is disabled in configuration`
- `// In fullscreen we must display controls` 

I was able to hide default controls on full screen by tweaking a bit but can we show custom controls on the default full screen @freeboub?

**Comment by freeboub** [comment]
Yes , as stated in doc, when you go to fullscreen control will be shown on ios.
The fullscreen is managed in a dedicated panel so you will never be able to display react ui over it.
I think you need to change the philosophy or your app to handle the use case you describe...

**Comment by nicolaosm** [comment]
So how can we have custom controls UI in iOS fullscreen? @freeboub

**Comment by nicolaosm** [comment]
> I just open PR which shall fix the issue here : #3230
> Can you please test it on your side ?

Yes, issue fixed on iOS @freeboub . Now app isn't crashing anymore and fullscreen is being presented.

**Comment by freeboub** [comment]
Just by security, I ensure this is the same behavior than 5.2.1 release.
Thank you for the test !

**Comment by freeboub** [comment]
> So how can we have custom controls UI in iOS fullscreen? @freeboub

on our mobile project we handle rotation manually (with react-native-orientation-locker I think). and adapt the UI depending of the orientation

**Comment by freeboub** [comment]
fix will be integrated in 6.0.0-alpha.8

**Comment by nicolaosm** [comment]
> on our mobile project we handle rotation manually (with react-native-orientation-locker I think). and adapt the UI depending of the orientation

Thanks @freeboub! So basically, when user press on full screen, you rotate the video itself to landscape and give it full height and width of the screen?
